kratos/spatial_containers/bins_dynamic.h

```diff
@@ -36,232 +52,303 @@ std::size_t TDimension,
+     * and sets the number of cells to 0.
+     */
+    BinsDynamic()
+        : mitPointsBegin(this->NullIterator()), mitPointsEnd(this->NullIterator()), mNumCells(0)
     {};
 
-    //************************************************************************
-
-    BinsDynamic( IteratorType const& PointBegin, IteratorType const& PointEnd, SizeType BucketSize = 1 )
-        : mPointBegin(PointBegin), mPointEnd(PointEnd)
+    /**
+     * @brief Constructor for BinsDynamic with iterators and optional bucket size.
+     * @param PointBegin   Iterator pointing to the beginning of the point data.
+     * @param PointEnd     Iterator pointing to the end of the point data.
+     * @param BucketSize   Optional parameter to specify the bucket size (default is 1).
+     */
+    BinsDynamic(IteratorType const& PointBegin, IteratorType const& PointEnd, SizeType BucketSize = 1)
+        : mitPointsBegin(PointBegin), mitPointsEnd(PointEnd)
     {
-        if(mPointBegin==mPointEnd)
+        if (mitPointsBegin == mitPointsEnd)
             return;
-        mNumCells = std::distance(mPointBegin,mPointEnd);
+        mNumCells = std::distance(mitPointsBegin, mitPointsEnd);
         CalculateBoundingBox();
         CalculateCellSize(mNumCells);
         AllocateCellsContainer();
         GenerateBins();
     }
 
-    //************************************************************************
-
-    BinsDynamic( IteratorType const& PointBegin, IteratorType const& PointEnd, PointType const& MinPoint, PointType const& MaxPoint, SizeType BucketSize = 1 )
-        : mPointBegin(PointBegin), mPointEnd(PointEnd)
+    /**
+     * @brief Constructor for BinsDynamic with iterators, min point, max point, and optional bucket size.
+     * @param PointBegin   Iterator pointing to the beginning of the point data.
+     * @param PointEnd     Iterator pointing to the end of the point data.
+     * @param MinPoint     The minimum point of the bounding box.
+     * @param MaxPoint     The maximum point of the bounding box.
+     * @param BucketSize   Optional parameter to specify the bucket size (default is 1).
+     */
+    BinsDynamic(IteratorType const& PointBegin, IteratorType const& PointEnd, PointType const& MinPoint, PointType const& MaxPoint, SizeType BucketSize = 1)
+        : mitPointsBegin(PointBegin), mitPointsEnd(PointEnd)
     {
-        if(mPointBegin==mPointEnd)
+        if (mitPointsBegin == mitPointsEnd)
             return;
 
-        mNumCells = std::distance(mPointBegin,mPointEnd);
-        for(SizeType i = 0 ; i < Dimension ; i++)
-        {
-            mMinPoint[i] = MinPoint[i];
-            mMaxPoint[i] = MaxPoint[i];
+        mNumCells = std::distance(mitPointsBegin, mitPointsEnd);
+        auto& r_min_point = GetMinPoint();
+        auto& r_max_point = GetMaxPoint();
+        for (SizeType i = 0; i < Dimension; i++) {
+            r_min_point[i] = MinPoint[i];
+            r_max_point[i] = MaxPoint[i];
```

Comment:
```c++
BinsDynamic(IteratorType const& PointBegin, IteratorType const& PointEnd, PointType const& MinPoint, PointType const& MaxPoint, SizeType BucketSize = 1)
        : mitPointsBegin(PointBegin), mitPointsEnd(PointEnd), mBoundingBox(MinPoint, MaxPoint)
    {
        if (mitPointsBegin == mitPointsEnd)
            return;

        mNumCells = std::distance(mitPointsBegin, mitPointsEnd);
 ```

Reply:
okay

Reply:
Again due to evil nature of intrussive pointers I cannot use this, and contrary to the point constructor it is not possible to modify the BB beacuse there is no common interface. I will keep it as it is.